.github/workflows/smoke-tests.yml

```diff
@@ -20,7 +20,7 @@ jobs:
   test:
     runs-on: ubuntu-latest
     env:
-      GH_TOKEN: ${{ github.token }}
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I'm afraid this is just another syntax to get the same token.

We would need to grant the token the correct [permissions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/assigning-permissions-to-jobs).

However, unfortunately, I don't see appropriate permission to query the GH user API.

Reply:
As an alternative, we could use a machine user token.

I don't know what the username is used for, but could we do something like:

```
if [[ -n "$CI"]]; then 
  user_name = "github-actions"
else 
# ...
fi
```
in https://github.com/elastic/apm-aws-lambda/blob/54ff53aaaefba7d67d93a71bbca14631749df3d5/testing/smoketest/get_username.sh#L11-L10 ?

Reply:
That's a good suggestion. Using a hard-coded user_name in CI should be fine.